### DISCUSSION TITLE
Don't generate a mutation for `.with_capacity`

### DISCUSSION THREAD

**Discussion by pashadia** [sourcefrog/cargo-mutants#314] [Ideas]
Hi all,

New contributor here, first of all congratulations on a great project!

I've used `cargo mutants` on a fairly large codebase (12kloc). One of the things that appear regularly on my project is to preallocate memory for some data structures. For instance, we use things like `let v = Vec::with_capacity(a+b);`. Obviously, `cargo-mutants` is generating a mutation for the argument, into things like `a-b` or `a*b`. However, this won't break any tests, as it doesn't actually introduce a bug in functionality, only the memory allocation footprint may be different.

My proposal is to remove the generation of mutations for the `.with_capacity` parameter, for all the structs in `std::collections`, and perhaps not only. I would gladly attempt an implementation, if this proposal is well received.

**Comment by sourcefrog** [comment]
Sounds like a great idea, and probably not too hard to have.

This change ought to go into `visit.rs` and to also have a unit test in that module. You can see existing examples there where we skip `new` and `Default`: but this is a bit different: we don't want to primarily skip mutations of `with_capacity`, but rather not recurse into the arguments of any call to functions with that name. (If you get stuck please let me know here.)
